packages/automated_actions/automated_actions/db/models/_action.py

```diff
@@ -93,12 +95,43 @@ def set_final_state(
 
     @classmethod
     def find_by_owner(
-        cls: type[Self], username: str, status: ActionStatus | None = None
+        cls: type[Self],
+        username: str,
+        status: ActionStatus | None = None,
+        max_age: int | None = None,
     ) -> Iterable[Action]:
         """Returns actions for owner."""
-        if status is None:
-            return cls.owner_index.query(username)
-        return cls.owner_index.query(username, cls.status == status.value)
+        match (status, max_age):
+            case (None, None):
+                # If no status or max_age is provided, return all actions for the user
+                return cls.owner_index.query(username)
+            case (None, max_age_val) if max_age_val is not None:
+                # If no status is provided, but max_age is, return actions
+                # for the user not older than max_age
+                return cls.owner_index.query(
+                    username,
+                    range_key_condition=cls.updated_at
+                    >= int(dt.now(tz=UTC).timestamp() - max_age_val),
```

Comment:
I checked index in dynamodb, range key is `status`

Reply:
I will manually migrate (delete and recreate) the existing tables.